app/src/main/java/com/sci/coffeeandroid/feature/menudetails/ui/screen/MenuDetailsActivity.kt

```diff
@@ -1,24 +1,45 @@
 package com.sci.coffeeandroid.feature.menudetails.ui.screen
 
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
 import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.AdapterView
 import android.widget.ArrayAdapter
-import android.widget.Spinner
+import android.widget.Toast
 import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.content.res.AppCompatResources
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
+import androidx.core.widget.addTextChangedListener
 import com.bumptech.glide.Glide
+import com.sci.coffeeandroid.MainActivity
 import com.sci.coffeeandroid.R
 import com.sci.coffeeandroid.databinding.ActivityMenuDetailsBinding
-import com.sci.coffeeandroid.feature.menudetails.ui.viewmodel.CoffeeDetailViewModel
-import kotlinx.coroutines.delay
+import com.sci.coffeeandroid.feature.home.HomeActivity
+import com.sci.coffeeandroid.feature.menudetails.domain.model.Size
+import com.sci.coffeeandroid.feature.menudetails.domain.model.Sugar
+import com.sci.coffeeandroid.feature.menudetails.domain.model.Variation
+import com.sci.coffeeandroid.feature.menudetails.ui.viewmodel.MenuDetailsViewModel
 import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
 
-class MenuDetailsActivity() : AppCompatActivity() {
+class MenuDetailsActivity : AppCompatActivity() {
 
-    private val coffeeDetailViewModel: CoffeeDetailViewModel by viewModel()
+    private val menuDetailsViewModel: MenuDetailsViewModel by viewModel() {
+        parametersOf(
+            coffeeId
+        )
+    }
     private lateinit var binding: ActivityMenuDetailsBinding
+    private val coffeeId: Int by lazy {
+        intent.getIntExtra("coffeeId", 0)
```

Comment:
should use val COFFEE_ID and the caller should be called MenuDetailsActivity.COFFEE_ID